# Análise de Dados -  Índice de desigualdade de gênero por país

Link: https://www.kaggle.com/datasets/iamsouravbanerjee/gender-inequality-index-dataset?resource=download



Insira o nome do grupo (no máximo 3 integrantes):  
1)  TIAGO PEREIRA DA SILVA - 23012171
2)  VINICIUS MENDES DA CUNHA - 23015801
3)  ENZO CINTO QUATROCHI - 23015904

Coloque abaixo a importação de todas as bibliotecas necessárias:

In [43]:
import pandas as pd
import numpy as np

1) Faça a leitura do arquivo "Gender Inequality Index.csv" como um dataframe

In [44]:
df = pd.read_csv("Gender Inequality Index.csv")

2) Apresente as 10 primeiras linhas do arquivo:

In [45]:
df.head(10)

ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   
5  ARG             Argentina   America  Southern Hemisphere   
6  ARM               Armenia      Asia  Northern Hemisphere   
7  ATG   Antigua and Barbuda   America  Northern Hemisphere   
8  AUS             Australia   Oceania  Southern Hemisphere   
9  AUT               Austria    Europe  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   
5                Very High                     LAC             47.0   
6                     High                     ECA             85.0   
7                     High                     LAC             71.0   
8                Very High                     NaN              5.0   
9                Very High                     NaN             25.0   

   GII Rank (2021)  Gender Inequality Index (1990)  \
0            167.0                             NaN   
1            136.0                           0.725   
2             39.0                             NaN   
3              NaN                             NaN   
4             11.0                           0.659   
5             69.0                           0.442   
6             53.0                           0.470   
7              NaN                             NaN   
8             19.0                           0.185   
9             12.0                           0.202   

   Gender Inequality Index (1991)  ...  Gender Inequality Index (2012)  \
0                             NaN  ...                           0.738   
1                           0.723  ...                           0.545   
2                             NaN  ...                           0.235   
3                             NaN  ...                             NaN   
4                           0.647  ...                           0.171   
5                           0.439  ...                           0.360   
6                           0.468  ...                           0.327   
7                             NaN  ...                             NaN   
8                           0.184  ...                           0.130   
9                           0.204  ...                           0.098   

   Gender Inequality Index (2013)  Gender Inequality Index (2014)  \
0                           0.728                           0.718   
1                           0.540                           0.531   
2                           0.225                           0.219   
3                             NaN                             NaN   
4                           0.161                           0.151   
5                           0.356                           0.351   
6                           0.305                           0.313   
7                             NaN                             NaN   
8                           0.123                           0.117   
9                           0.086                           0.083   

   Gender Inequality Index (2015)  Gender Inequality Index (2016)  \
0                           0.706                           0.692   
1                           0.530                           0.529   
2                           0.204                           0.191   
3                             NaN                             NaN   
4          

3) Qual o formato do DataFrame? (Ou seja, quantas linhas e quantas colunas ele possui?)

In [46]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 40 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   ISO3                            195 non-null    object 
 1   Country                         195 non-null    object 
 2   Continent                       195 non-null    object 
 3   Hemisphere                      195 non-null    object 
 4   Human Development Groups        191 non-null    object 
 5   UNDP Developing Regions         151 non-null    object 
 6   HDI Rank (2021)                 191 non-null    float64
 7   GII Rank (2021)                 170 non-null    float64
 8   Gender Inequality Index (1990)  128 non-null    float64
 9   Gender Inequality Index (1991)  128 non-null    float64
 10  Gender Inequality Index (1992)  128 non-null    float64
 11  Gender Inequality Index (1993)  128 non-null    float64
 12  Gender Inequality Index (1994)  128 

O dataframe tem o formato de uma matriz, que possui 195 linhas e 40 colunas.

4) Quais são os nomes de todas as colunas do dataframe?

In [47]:
df.columns

Index(['ISO3', 'Country', 'Continent', 'Hemisphere',
       'Human Development Groups', 'UNDP Developing Regions',
       'HDI Rank (2021)', 'GII Rank (2021)', 'Gender Inequality Index (1990)',
       'Gender Inequality Index (1991)', 'Gender Inequality Index (1992)',
       'Gender Inequality Index (1993)', 'Gender Inequality Index (1994)',
       'Gender Inequality Index (1995)', 'Gender Inequality Index (1996)',
       'Gender Inequality Index (1997)', 'Gender Inequality Index (1998)',
       'Gender Inequality Index (1999)', 'Gender Inequality Index (2000)',
       'Gender Inequality Index (2001)', 'Gender Inequality Index (2002)',
       'Gender Inequality Index (2003)', 'Gender Inequality Index (2004)',
       'Gender Inequality Index (2005)', 'Gender Inequality Index (2006)',
       'Gender Inequality Index (2007)', 'Gender Inequality Index (2008)',
       'Gender Inequality Index (2009)', 'Gender Inequality Index (2010)',
       'Gender Inequality Index (2011)', 'Gender Inequal

5) Renomeie as colunas conforme lista abaixo:
* Country -> País
* Continent -> Continente
* Hemisphere -> Hemisfério
* Human Development Groups -> Grupo de Desenvolvimento Humano
* UNDP Developing Regions -> Regiões em desenvolvimento do UNDP
* HDI Rank (2021) -> IDH Rank (2021)
* GII Rank (2021) -> IDG Rank (2021)

In [48]:
df.rename(columns={"Country":"País",
                   "Continent":"Continente",
                   "Hemisphere":"Hemisfério",
                   "Human Development Groups":"Grupo de Desenvolvimento Humano",
                   "UNDP Developing Regions":"Regiões em desenvolvimento do UNDP",
                   "HDI Rank (2021)":"IDH Rank (2021)",
                   "GII Rank (2021)":"IDG Rank (2021)",}, inplace=True)

6) Acesse o registro da linha de indice 123.

In [49]:
df.iloc[123]

ISO3                                                  MWI
País                                               Malawi
Continente                                         Africa
Hemisfério                            Southern Hemisphere
Grupo de Desenvolvimento Humano                       Low
Regiões em desenvolvimento do UNDP                    SSA
IDH Rank (2021)                                     169.0
IDG Rank (2021)                                     142.0
Gender Inequality Index (1990)                      0.718
Gender Inequality Index (1991)                      0.717
Gender Inequality Index (1992)                      0.714
Gender Inequality Index (1993)                      0.711
Gender Inequality Index (1994)                       0.71
Gender Inequality Index (1995)                      0.708
Gender Inequality Index (1996)                      0.706
Gender Inequality Index (1997)                      0.705
Gender Inequality Index (1998)                      0.704
Gender Inequal

7) Acesse as informações da coluna "IDH Rank (2021)"

In [50]:
df["IDH Rank (2021)"]

0      180.0
1      148.0
2       67.0
3       40.0
4       26.0
       ...  
190    111.0
191    183.0
192    109.0
193    154.0
194    146.0
Name: IDH Rank (2021), Length: 195, dtype: float64

8) Encontre as informações referentes ao Brasil (Dica: O nome do país está como "Brazil")

In [51]:
df[df["País"] == "Brazil"]

ISO3    País Continente           Hemisfério  \
23  BRA  Brazil    America  Southern Hemisphere   

   Grupo de Desenvolvimento Humano Regiões em desenvolvimento do UNDP  \
23                            High                                LAC   

    IDH Rank (2021)  IDG Rank (2021)  Gender Inequality Index (1990)  \
23             87.0             94.0                           0.552   

    Gender Inequality Index (1991)  ...  Gender Inequality Index (2012)  \
23                           0.549  ...                           0.444   

    Gender Inequality Index (2013)  Gender Inequality Index (2014)  \
23                           0.444                            0.44   

    Gender Inequality Index (2015)  Gender Inequality Index (2016)  \
23                           0.437                           0.431   

    Gender Inequality Index (2017)  Gender Inequality Index (2018)  \
23                           0.421                           0.401   

    Gender Inequality Index (2019)  Gender Inequality Index (2020)  \
23                           0.395                           0.395   

    Gender Inequality Index (2021)  
23                            0.39  

[1 rows x 40 columns]

9) Transforme o Nome do País como Index do Dataframe. Em seguida, remova a coluna País.

In [52]:
df['index'] = df['País']
df.set_index('index', inplace=True)
df.drop("País", axis=1, inplace=True)
df.head()


ISO3 Continente           Hemisfério  \
index                                                       
Afghanistan           AFG       Asia  Northern Hemisphere   
Angola                AGO     Africa  Southern Hemisphere   
Albania               ALB     Europe  Northern Hemisphere   
Andorra               AND     Europe  Northern Hemisphere   
United Arab Emirates  ARE       Asia  Northern Hemisphere   

                     Grupo de Desenvolvimento Humano  \
index                                                  
Afghanistan                                      Low   
Angola                                        Medium   
Albania                                         High   
Andorra                                    Very High   
United Arab Emirates                       Very High   

                     Regiões em desenvolvimento do UNDP  IDH Rank (2021)  \
index                                                                      
Afghanistan                                          SA            180.0   
Angola                                              SSA            148.0   
Albania                                             ECA             67.0   
Andorra                                             NaN             40.0   
United Arab Emirates                                 AS             26.0   

                      IDG Rank (2021)  Gender Inequality Index (1990)  \
index                                                                   
Afghanistan                     167.0                             NaN   
Angola                          136.0                           0.725   
Albania                          39.0                             NaN   
Andorra                           NaN                             NaN   
United Arab Emirates             11.0                           0.659   

                      Gender Inequality Index (1991)  \
index                                                  
Afghanistan                                      NaN   
Angola                                         0.723   
Albania                                          NaN   
Andorra                                          NaN   
United Arab Emirates                           0.647   

                      Gender Inequality Index (1992)  ...  \
index                                                 ...   
Afghanistan                                      NaN  ...   
Angola                                         0.721  ...   
Albania                                          NaN  ...   
Andorra                                          NaN  ...   
United Arab Emirates                           0.640  ...   

                      Gender Inequality Index (2012)  \
index                                                  
Afghanistan                                    0.738   
Angola                                         0.545   
Albania                                        0.235   
Andorra                                          NaN   
United Arab Emirates                           0.171   

                      Gender Inequality Index (2013)  \
index                                                  
Afghanistan                                    0.728   
Angola                                         0.540   
Albania                                        0.225   
Andorra                                          NaN   
United Arab Emirates                           0.161   

                      Gender Inequality Index (2014)  \
index                                                  
Afghanistan                                    0.718   
Angola                                         0.531   
Albania                                        0.219   
Andorra                                          NaN   
United Arab Emirates                           0.151   

                      Gender Inequality Index (2015)  \
index                                                  
Afghanistan                                

10) Acesse as informações da Argentina e do Brasil em um único acesso. 

In [53]:
df.loc[["Brazil", "Argentina"]]

ISO3 Continente           Hemisfério  \
index                                            
Brazil     BRA    America  Southern Hemisphere   
Argentina  ARG    America  Southern Hemisphere   

          Grupo de Desenvolvimento Humano Regiões em desenvolvimento do UNDP  \
index                                                                          
Brazil                               High                                LAC   
Argentina                       Very High                                LAC   

           IDH Rank (2021)  IDG Rank (2021)  Gender Inequality Index (1990)  \
index                                                                         
Brazil                87.0             94.0                           0.552   
Argentina             47.0             69.0                           0.442   

           Gender Inequality Index (1991)  Gender Inequality Index (1992)  \
index                                                                       
Brazil                              0.549                           0.545   
Argentina                           0.439                           0.436   

           ...  Gender Inequality Index (2012)  \
index      ...                                   
Brazil     ...                           0.444   
Argentina  ...                           0.360   

           Gender Inequality Index (2013)  Gender Inequality Index (2014)  \
index                                                                       
Brazil                              0.444                           0.440   
Argentina                           0.356                           0.351   

           Gender Inequality Index (2015)  Gender Inequality Index (2016)  \
index                                                                       
Brazil                              0.437                           0.431   
Argentina                           0.345                           0.338   

           Gender Inequality Index (2017)  Gender Inequality Index (2018)  \
index                                                                       
Brazil                              0.421                           0.401   
Argentina                           0.328                           0.315   

           Gender Inequality Index (2019)  Gender Inequality Index (2020)  \
index                                                                       
Brazil                              0.395                           0.395   
Argentina                           0.306                           0.293   

           Gender Inequality Index (2021)  
index                                      
Brazil                              0.390  
Argentina                           0.287  

[2 rows x 39 columns]

11) Quais os diferentes valores (valores únicos) que aparecem nas colunas Continente, Hemisfério, Grupo de Desenvolvimento Humano e Regiões em desenvolvimento do UNDP

In [54]:
df["Continente"].unique()

array(['Asia', 'Africa', 'Europe', 'America', 'Oceania'], dtype=object)

In [55]:
df["Regiões em desenvolvimento do UNDP"].unique()

array(['SA', 'SSA', 'ECA', nan, 'AS', 'LAC', 'EAP'], dtype=object)

In [56]:
df["Grupo de Desenvolvimento Humano"].unique()

array(['Low', 'Medium', 'High', 'Very High', nan], dtype=object)

In [57]:
df["Hemisfério"].unique()

array(['Northern Hemisphere', 'Southern Hemisphere'], dtype=object)

12) Altere todos os "Northern Hemisphere" por "Norte" e todos os "Southern Hemisphere" por "Sul" 

In [58]:
df["Hemisfério"] = df["Hemisfério"].replace({
    "Northern Hemisphere": "Norte",
    "Southern Hemisphere": "Sul",
})

df["Hemisfério"]

index
Afghanistan             Norte
Angola                    Sul
Albania                 Norte
Andorra                 Norte
United Arab Emirates    Norte
                        ...  
Samoa                     Sul
Yemen                   Norte
South Africa              Sul
Zambia                    Sul
Zimbabwe                  Sul
Name: Hemisfério, Length: 195, dtype: object

13) Faça a atualização de todos os continentes para o português. Utilize um dicionário com a função replace.

14) Verifique quantos e quais são os registros que contém valor nulo na coluna "Grupo de Desenvolvimento Humano"

In [59]:
df[df["Grupo de Desenvolvimento Humano"].isnull()]

ISO3 Continente Hemisfério Grupo de Desenvolvimento Humano  \
index                                                                    
Monaco       MCO     Europe      Norte                             NaN   
Nauru        NRU    Oceania        Sul                             NaN   
North Korea  PRK       Asia      Norte                             NaN   
Somalia      SOM     Africa      Norte                             NaN   

            Regiões em desenvolvimento do UNDP  IDH Rank (2021)  \
index                                                             
Monaco                                     NaN              NaN   
Nauru                                      EAP              NaN   
North Korea                                EAP              NaN   
Somalia                                     AS              NaN   

             IDG Rank (2021)  Gender Inequality Index (1990)  \
index                                                          
Monaco                   NaN                             NaN   
Nauru                    NaN                             NaN   
North Korea              NaN                             NaN   
Somalia                  NaN                             NaN   

             Gender Inequality Index (1991)  Gender Inequality Index (1992)  \
index                                                                         
Monaco                                  NaN                             NaN   
Nauru                                   NaN                             NaN   
North Korea                             NaN                             NaN   
Somalia                                 NaN                             NaN   

             ...  Gender Inequality Index (2012)  \
index        ...                                   
Monaco       ...                             NaN   
Nauru        ...                             NaN   
North Korea  ...                             NaN   
Somalia      ...                             NaN   

             Gender Inequality Index (2013)  Gender Inequality Index (2014)  \
index                                                                         
Monaco                                  NaN                             NaN   
Nauru                                   NaN                             NaN   
North Korea                             NaN                             NaN   
Somalia                                 NaN                             NaN   

             Gender Inequality Index (2015)  Gender Inequality Index (2016)  \
index                                                                         
Monaco                                  NaN                             NaN   
Nauru                                   NaN                             NaN   
North Korea                             NaN                             NaN   
Somalia                                 NaN                             NaN   

             Gender Inequality Index (2017)  Gender Inequality Index (2018)  \
index                                                                         
Monaco                                  NaN                             NaN   
Nauru                                   NaN                             NaN   
North Korea                             NaN                             NaN   
Somalia                                 NaN                             NaN   

             Gender Inequality Index (2019)  Gender Inequality Index (2020)  \
index                                                                         
Monaco                                  NaN                             NaN   
Nauru                                   NaN                             NaN   
North Korea                             NaN                             NaN   
Somalia                                 NaN                             NaN   

             Gender Inequality Index (2021)  
index                                        

A coluna possui 4 valores nulos em Monaco, Nauru , Coreia do Norte e Somalia.

15) Remova os registros selecionados na questão anterior.

In [60]:
df = df[df["Grupo de Desenvolvimento Humano"].notnull()]

16) O dataset ainda possuí valores nulos? 

In [61]:
df.isnull().sum()

ISO3                                   0
Continente                             0
Hemisfério                             0
Grupo de Desenvolvimento Humano        0
Regiões em desenvolvimento do UNDP    43
IDH Rank (2021)                        0
IDG Rank (2021)                       21
Gender Inequality Index (1990)        63
Gender Inequality Index (1991)        63
Gender Inequality Index (1992)        63
Gender Inequality Index (1993)        63
Gender Inequality Index (1994)        63
Gender Inequality Index (1995)        61
Gender Inequality Index (1996)        61
Gender Inequality Index (1997)        61
Gender Inequality Index (1998)        55
Gender Inequality Index (1999)        50
Gender Inequality Index (2000)        47
Gender Inequality Index (2001)        42
Gender Inequality Index (2002)        41
Gender Inequality Index (2003)        38
Gender Inequality Index (2004)        35
Gender Inequality Index (2005)        32
Gender Inequality Index (2006)        29
Gender Inequalit

Sim, o dataset ainda possui diversos valores nulos.

17) Ordene os valores pela coluna "Gender Inequality Index (2021)" e responda: quem são os top 5 países nesse ranking? (Ou seja, os países que possuem a menor desigualdade)

In [62]:
df["Gender Inequality Index (2021)"].sort_values(ascending= True)

index
Denmark            0.013
Norway             0.016
Switzerland        0.018
Sweden             0.023
Netherlands        0.025
                   ...  
Solomon Islands      NaN
San Marino           NaN
Seychelles           NaN
Tuvalu               NaN
Vanuatu              NaN
Name: Gender Inequality Index (2021), Length: 191, dtype: float64

Os países com menor desigualdade são: Dinamarca, Finlândia, Suíça e Países Baixos.

18) E os cinco últimos colocados, quais seriam? Desconsidere os valores Nulos. Ou seja, os últimos colocados que de fato foram calculados.

In [63]:
df["Gender Inequality Index (2021)"].sort_values(ascending= False)

index
Yemen                       0.820
Papua New Guinea            0.725
Nigeria                     0.680
Afghanistan                 0.678
Central African Republic    0.672
                            ...  
Solomon Islands               NaN
San Marino                    NaN
Seychelles                    NaN
Tuvalu                        NaN
Vanuatu                       NaN
Name: Gender Inequality Index (2021), Length: 191, dtype: float64

Os países com maior desigualdade são: Yemen, Papua Nova Guiné, Nigéria, Afeganistão e Republica Central Africana.

19) Apresente a média de Gender Inequality Index (2021)  dos países da Região em desenvolvimento do UNDP - LAC (Latin America and the Caribbean)

In [64]:
lac = df[df["Regiões em desenvolvimento do UNDP"] == "LAC"]
lac["Gender Inequality Index (2021)"].mean()

np.float64(0.37751724137931036)

20) Apresente a média de Gender Inequality Index (2021) para cada Região em desenvolvimento do UNDP (utilize o comando group by)

In [66]:
df.groupby("Regiões em desenvolvimento do UNDP")["Gender Inequality Index (2021)"].mean()

Regiões em desenvolvimento do UNDP
AS     0.382235
EAP    0.378294
ECA    0.203235
LAC    0.377517
SA     0.476556
SSA    0.552452
Name: Gender Inequality Index (2021), dtype: float64

21) Qual o nome do país com o menor índice de desigualdade no ano de 2021 por continente? 

22) Verifique a posição do IDH Rank (2021) dos países que você obteve como resposta no exercício anterior. 

23) Existe uma correlação entre o 'IDH Rank (2021)' e o 'IDG Rank (2021)'? Interprete o resultado. 

24) Aplique o cálculo da distância interquartílica para determinar se existem valores outliers na coluna "Gender Inequality Index (2021)" Lembre-se que o cálculo da distância interquartílica é dado por: 

IQR = q3 - q1

Limite_inferior = q1 - 1.5*IQR

Limite_superior = q3 + 1.5*IQR

25) Faça o gráfico bloxplot da coluna Gender Inequality Index (2021) e verifique a resposta do exercício anterior. 

26) Faça o histograma para "Gender Inequality Index (2021)" e responda: os dados estão distribuidos no formato da distruição normal?

27) Entre os anos de 1990 e 2021, qual foi o país que mais evoluí no índice? Considere apenas a listagem dos 128 países disponíveis em 1990.

28) Faça um gráfico de linha que apresente duas séries:

    -A primeira representa a  evolução do Brasil no período de 1990 a 2021
    -A segunda série representa a média do indice dos paises da LAC.
    
    Eixo X - Anos
    Eixo Y - Índice.

29) Escolha um país a sua escolha e faça um gráfico de linha para apresentar a evolução do índice (Similar ao Brasil no exercício anterior). A diferença que desta vez os dados deverão estar rotulados. Dica: Utilize a função Annotate.

30) Faça um gráfico de barras para indicar a quantidade de países em cada continente que contém classificado grupo de desenvolvimento Very High. 

31) Faça um gráfico de pizza para representar a quantidade de países em cada hemisfério.

32) Faça um gráfico de dispersão onde o eixo X representa o IDH Rank, o eixo Y representa o IDG Rank e os marcadores representarão os continentes dos países. Escolha uma cor e um marcador diferente para cada continente.


33) Faça uma análise dos dados e gere um gráfico a sua escolha. 

34) Faça uma análise dos dados e gere um gráfico a sua escolha. 

35) Crie um arquivo .csv para salvar as informações dos países da Ásia em um arquivo separado. 
A primeira linha do arquivo deverá conter o nome de todas as colunas. 
As colunas presente no arquivo deverão ser: Nome, ISO3,Continente,Hemisfério,Grupo de Desenvolvimento Humano,Regiões em desenvolvimento do UNDP,IDH Rank (2021),IDG Rank (2021)